ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/ListIndexTool.java

```diff
@@ -96,23 +110,25 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        // TODO: This logic exactly matches the OpenSearch _cat/indices REST action. If code at
-        // o.o.rest/action/cat/RestIndicesAction.java changes those changes need to be reflected here
-        // https://github.com/opensearch-project/ml-commons/pull/1582#issuecomment-1796962876
+        // TODO: This logic exactly matches the OpenSearch _list/indices REST action. If code at
+        // o.o.rest/action/list/RestIndicesListAction.java changes those changes need to be reflected here
```

Comment:
how are we planning to track changes of `o.o.rest/action/list/RestIndicesListAction.java` to reflect here?

Reply:
It's not easy to keep tracking of that, maybe sometime we encounter error in the IT then we can realize that OS core has made changes that we should reflect, if the change is not critical then we may not find any issue with this tool, in this case I think it's fine not to track the changes. BTW, I have a plan to support dynamic tool in the future so that we don't need to duplicate the code of OpenSearch core or other plugins, you can check here: https://github.com/opensearch-project/skills/issues/459